jraft-extension/rpc-grpc-impl/src/main/java/com/alipay/sofa/jraft/rpc/impl/GrpcServer.java

```diff
@@ -103,48 +141,93 @@ public void registerProcessor(final RpcProcessor processor) {
 
         final ServerCallHandler<Message, Message> handler = ServerCalls.asyncUnaryCall(
                 (request, responseObserver) -> {
-                    final SocketAddress remoteAddress = RemoteAddressInterceptor.REMOTE_ADDRESS.get();
+                    final SocketAddress remoteAddress = RemoteAddressInterceptor.getRemoteAddress();
+                    final Connection conn = ConnectionInterceptor.getCurrentConnection(this.closedEventListeners);
 
                     final RpcContext rpcCtx = new RpcContext() {
 
                         @Override
                         public void sendResponse(final Object responseObj) {
-                            responseObserver.onNext((Message) responseObj);
-                            responseObserver.onCompleted();
+                            try {
+                                responseObserver.onNext((Message) responseObj);
+                                responseObserver.onCompleted();
+                            } catch (final Throwable t) {
+                                LOG.warn("[GRPC] failed to send response: {}.", t);
+                            }
                         }
 
                         @Override
                         public Connection getConnection() {
-                            throw new UnsupportedOperationException("unsupported");
+                            if (conn == null) {
+                                throw new UnsupportedOperationException("unsupported");
```

Comment:
UnsupportedOperationException?   Maybe `IllegalStateException` is more proble.

Reply:
done